packages/x-data-grid-pro/src/hooks/features/rowReorder/utils.ts

```diff
@@ -1,119 +1,49 @@
-export const conditions = {
-  // Node type checks
-  isGroupToGroup: (ctx: Ctx) => ctx.sourceNode.type === 'group' && ctx.targetNode.type === 'group',
-
-  isLeafToLeaf: (ctx: Ctx) => ctx.sourceNode.type === 'leaf' && ctx.targetNode.type === 'leaf',
-
-  isLeafToGroup: (ctx: Ctx) => ctx.sourceNode.type === 'leaf' && ctx.targetNode.type === 'group',
-
-  isGroupToLeaf: (ctx: Ctx) => ctx.sourceNode.type === 'group' && ctx.targetNode.type === 'leaf',
-
-  // Drop position checks
-  isDropAbove: (ctx: Ctx) => ctx.dropPosition === 'above',
-  isDropBelow: (ctx: Ctx) => ctx.dropPosition === 'below',
-
-  // Depth checks
-  sameDepth: (ctx: Ctx) => ctx.sourceNode.depth === ctx.targetNode.depth,
-
-  sourceDepthGreater: (ctx: Ctx) => ctx.sourceNode.depth > ctx.targetNode.depth,
-
-  targetDepthIsSourceMinusOne: (ctx: Ctx) => ctx.targetNode.depth === ctx.sourceNode.depth - 1,
-
-  // Parent checks
-  sameParent: (ctx: Ctx) => ctx.sourceNode.parent === ctx.targetNode.parent,
-
-  // Node state checks
-  targetGroupExpanded: (ctx: Ctx) =>
-    (ctx.targetNode.type === 'group' && (ctx.targetNode as GridGroupNode).childrenExpanded) ??
-    false,
-
-  targetGroupCollapsed: (ctx: Ctx) =>
-    ctx.targetNode.type === 'group' && !(ctx.targetNode as GridGroupNode).childrenExpanded,
-
-  // Previous/Next node checks
-  hasPrevNode: (ctx: Ctx) => ctx.prevNode !== null,
-  hasNextNode: (ctx: Ctx) => ctx.nextNode !== null,
-
-  prevIsLeaf: (ctx: Ctx) => ctx.prevNode?.type === 'leaf',
-  prevIsGroup: (ctx: Ctx) => ctx.prevNode?.type === 'group',
-  nextIsLeaf: (ctx: Ctx) => ctx.nextNode?.type === 'leaf',
-  nextIsGroup: (ctx: Ctx) => ctx.nextNode?.type === 'group',
-
-  prevDepthEquals: (ctx: Ctx, depth: number) => ctx.prevNode?.depth === depth,
-
-  prevDepthEqualsSource: (ctx: Ctx) => ctx.prevNode?.depth === ctx.sourceNode.depth,
+export function findCellElement(target: EventTarget | null): Element {
+  const element = target as Element;
+  if (!element) {
+    return element;
+  }
 
-  // Complex checks
-  prevBelongsToSource: (ctx: Ctx) => {
-    if (!ctx.prevNode) {
-      return false;
-    }
-    // Check if prevNode.parent OR any of its ancestors === sourceNode.id
-    let currentId = ctx.prevNode.parent;
-    while (currentId) {
-      if (currentId === ctx.sourceNode.id) {
-        return true;
-      }
-      const node = ctx.rowTree[currentId];
-      if (!node) {
-        break;
-      }
-      currentId = node.parent;
-    }
-    return false;
-  },
-
-  // Position checks
-  isAdjacentPosition: (ctx: Ctx) => {
-    const { sourceRowIndex, targetRowIndex, dropPosition } = ctx;
-    return (
-      (dropPosition === 'above' && targetRowIndex === sourceRowIndex + 1) ||
-      (dropPosition === 'below' && targetRowIndex === sourceRowIndex - 1)
-    );
-  },
+  // Check if the target itself is a cell
+  if (
+    element instanceof Element &&
+    element.className &&
+    typeof element.className === 'string' &&
+    element.className.includes('cell')
```

Comment:
Should probably check for MuiDataGrid-cell specifically, no?


```suggestion
    element.classList.contains(gridClasses.cell)
```